### PR TITLE
Add inter branch merge flow config for 9.0 RC branches

### PR DESCRIPTION
**Pull request by carlossanlop**

### CODE REVIEW

File: github-merge-flow.jsonc [hunk 1]
```diff
@@ -15,6 +15,11 @@
         "release/8.0-rc":{
             "MergeToBranch": "release/8.0",
             "ExtraSwitches": "-QuietComments"
+        },
+        // Automate merging runtime release/9.0-rc branches back to release/9.0
+        "release/9.0-rc":{
```
Comment (mmitche) [inline github-merge-flow.jsonc:20]: I don't think are wildcarded like they were in old Maestro. Split into an rc1 and rc2 entry.

File: github-merge-flow.jsonc [hunk 2]
```diff
@@ -15,6 +15,11 @@
...
+            "MergeToBranch": "release/9.0",
+            "ExtraSwitches": "-QuietComments"
         }
```
Comment (carlossanlop) [inline github-merge-flow.jsonc:23]: Splitting as suggested:
```suggestion
        },
        // Automate merging runtime release/9.0-rc1 branch back to release/9.0
        "release/9.0-rc1":{
            "MergeToBranch": "release/9.0",
            "ExtraSwitches": "-QuietComments"
        },
        // Automate merging runtime release/9.0-rc2 branch back to release/9.0
        "release/9.0-rc2":{
            "MergeToBranch": "release/9.0",
            "ExtraSwitches": "-QuietComments"
        }
```

**Review by hoyosjs — APPROVED** [review]

**Review by lewing — APPROVED** [review]

**Comment by carlossanlop** [comment]
No CI validation for this change. Will merge right away to get this included in the 9.0 branch.

**Comment by carlossanlop** [comment]
This isn't working. We have had some new commits in the release/9.0-rc1 branch but I don't see any PRs created to autoflow those commits to the released/9.0 branch. What am I missing, @jkoritzinsky @mmitche @hoyosjs?

**Comment by carlossanlop** [comment]
Yeah, it's not running: https://github.com/dotnet/runtime/actions/workflows/inter-branch-merge-flow.yml

What I suspect is that it won't work unless it's in both release/9.0-rc1 and release/9.0.

I was hoping it would get autotriggered by just existing in release/9.0-rc1. I'll backport it to the remaining branch and see if that does it.

**Comment by carlossanlop** [comment]
Actually, I _snapped_ both release/9.0 and release/9.0-rc1, meaning all the necessary branches already include this.

**Comment by f-alizada** [comment]
@carlossanlop I have created the PR https://github.com/dotnet/runtime/pull/106614 which adds the workflow file to the branch. 
This is the reason it is not triggering on merges into the branches. 
Adding the workflow to the `main` here: https://github.com/dotnet/runtime/pull/106616 as well to avoid such situations in future.

**Pull request merged**